### PR TITLE
wayland-protocols 1.21

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **wayland-protocols/1.21**

close #6946

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
Here we go: this recipe only packages xml files, and it should generate the following .pc file:
```
prefix=/usr
datarootdir=${prefix}/share
pkgdatadir=${pc_sysrootdir}${datarootdir}/wayland-protocols

Name: Wayland Protocols
Description: Wayland protocol files
Version: 1.20
```
Which conan is currently not able to generate, because it has no way of specifying pkg-config variables.
We have to choose between:
- making it a regular conan package, add it to KB-H014 allow list, and let consumer recipe patch the .pc file like [this](https://github.com/bincrafters/conan-gtk/commit/2ce2d80b5c5a46148c265ca711be229aa788f922#diff-63f09721ade419d29886e8190ffa3c9c756baa9bef8802a7f2aeabf3b545164fR107-R108)
- making it a system package, but it's not very satisfying because we are able to compile it properly
- don't make a package for wayland-procols, which means gtk package ([regular package](https://github.com/conan-io/conan-center-index/pull/3624), not the system one currently available on CCI) will have to be built without wayland support by default
- something else ?

**Comment by ericLemanissier** [comment]
that's high level crickets!
@uilianries @SSE4 @Croydon wdyt ?

**Comment by Croydon** [comment]
Urgh o.O

Is there any chance that pkg-config files can get a path to themselves, so we can construct a path that always works?

**Comment by ericLemanissier** [comment]
I don't think so. the "regular package" solution would need something like https://github.com/conan-io/conan/issues/4048 in order to patch the .pc file after install, or manual patching by consumers like https://github.com/bincrafters/conan-gtk/commit/2ce2d80b5c5a46148c265ca711be229aa788f922#diff-63f09721ade419d29886e8190ffa3c9c756baa9bef8802a7f2aeabf3b545164fR107-R108

**Comment by Croydon** [comment]
I agree that this feels like it requires more Conan features.

Either we let Conan generate the `.pc` file and we get a correct `prefix` path to the package folder, but we won't have the custom variables then.

Or we package the `.pc` file and could define own variables, but there seems to be no way to guess the correct path after installation of the package.

**Comment by ericLemanissier** [comment]
Rethinking about this, I think the least bad option is to use a regular conan package, and rely on consumers (for now, that would be only gtk) to  patch the generated `wayland-protocols.pc` and add the correct path in the pkgdatadir variable

**Comment by ericLemanissier** [comment]
this is now blocked by the deployment of conan 1.36.0 now that https://github.com/conan-io/conan/pull/8727 is merged

**Comment by ericLemanissier** [comment]
blocked by https://github.com/conan-io/hooks/pull/337

**Review by prince-chrismc — APPROVED** [review]

File: recipes/wayland-protocols/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    _autotools = None
+
+    build_requires = (
+        "wayland/1.19.0"
+    )
```
Comment (jgsogo) [inline recipes/wayland-protocols/all/conanfile.py:26]: This is not needed to build (at least using Meson) although the build-system declares the dependency.

File: recipes/wayland-protocols/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+class WaylandProtocolsConan(ConanFile):
+    name = "wayland-protocols"
+    description = "The Wayland-Protocols package contains additional Wayland protocols that add functionality outside of protocols already in the Wayland core."
+    topics = ("conan", "wayland-protocols", "wayland")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://wayland.freedesktop.org/"
+    license = "MIT"
+    generators = "pkg_config"
```
Comment (jgsogo) [inline recipes/wayland-protocols/all/conanfile.py:12]: It has no deps

```suggestion
```

**Review by jgsogo — COMMENTED** [review]
I didn't realize about this PR and I've created mine here: https://github.com/conan-io/conan-center-index/pull/6946

I'm fine moving forward with any of them.

**Comment by ericLemanissier** [comment]
When this recipe was originally written, wayland-protocols did not have meson support. Let's continue with the other PR

**Pull request closed without merging**